src/Symfony/Component/Console/Application.php

```diff
@@ -466,7 +466,9 @@ public function get($name)
             $command = $this->commands[$name];
         } elseif ($this->commandLoader && $this->commandLoader->has($name)) {
             $command = $this->commandLoader->get($name);
-            $command->setName($name);
+            if (!$command->getName()) {
+                $command->setName($name);
+            }
             $this->add($command);
```

Comment:
Sounds reasonable :) It's just that it  will cache the command aliases, not sure that could cause WTF's?

https://github.com/chalasr/symfony/blob/f25a8b5c41ac9d2abf621f5990d4c59e7889df16/src/Symfony/Component/Console/Application.php#L447

Other then that :+1: